File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.dart
class Solution {
  int minSubArrayLen(int target, List<int> nums) {
    int l = 0;
    int sum = 0;
    int min_length = nums.length + 1;
    for (int r = 0; r < nums.length; ++r) {
      sum += nums[r];

      while (sum >= target) {
        min_length = min(min_length, (r - l + 1));
        sum -= nums[l];
        l += 1;
      }
    }

    return min_length == nums.length + 1 ? 0 : min_length;
  }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        l = 0
        sum = 0
        min_length = float('inf')
        n = len(nums)
        
        for r in range(n):
            sum += nums[r]
            
            while sum >= target:
                min_length = min(min_length, r-l+1)
                sum = sum - nums[l]
                l += 1
        
        return 0 if min_length == float('inf') else min_length

print(Solution().minSubArrayLen(7,[2,3,1,2,4,3]))